pkg/os/shell/shell_unix.go

```diff
@@ -6,20 +6,100 @@ package shell
 import (
 	"errors"
 	"fmt"
+	"os"
 	"path/filepath"
+
+	"github.com/shirou/gopsutil/v3/process"
+	"github.com/spf13/cast"
 )
 
 var (
 	ErrUnknownShell = errors.New("Error: Unknown shell")
+	getPid          = getCurrentProcess
 )
 
+// AbstractProcess is an interface created to abstract operations of the gopsutil library
+// It is created so that we can override the behavior while writing unit tests by providing
+// a mock implementation.
+type AbstractProcess interface {
+	Name() (string, error)
+	Parent() (AbstractProcess, error)
+}
+
+// RealProcess is a wrapper implementation of AbstractProcess to wrap around the gopsutil library's
+// process.Process object. This implementation is used in production code.
+type RealProcess struct {
+	realProcess *process.Process
+}
+
+func (p *RealProcess) Name() (string, error) {
+	return p.realProcess.Name()
+}
+
+func (p *RealProcess) Parent() (AbstractProcess, error) {
+	parentProcess, err := p.realProcess.Parent()
+	if err != nil {
+		return nil, err
+	}
+	return &RealProcess{parentProcess}, nil
+}
+
+func NewRealProcess(pid int32) (*RealProcess, error) {
```

Comment:
`getCurrentProcess` is the single user of `NewRealProcess`, you could consider merging the 2 especially as most of their code is error-handling.

Reply:
You're right. I've merged both methods into single one